AggregateMetrics/AggregateMetrics/Two/Implementation/HistogramImplementation.cs

```diff
@@ -60,6 +125,18 @@ public MetricTelemetry GetValueAndReset()
                     metric.Min = curMinValue;
                     metric.Max = curMaxValue;
                 }
+                else if (this.shouldCalculatePercentiles)
+                {
+                    PercentileAggregations percentiles = CalculatePercentiles();
```

Comment:
you'll need to swap list with the empty one here with interlocked operation so the list wouldn't be updated while you are doing calculations